salt/state.py

```diff
@@ -2731,10 +2731,13 @@ def verify_retry_data(self, retry_data):
                         ]
                 else:
                     validated_retry_data[expected_key] = retry_defaults[expected_key]
+
+        elif isinstance(retry_data, bool) and retry_data:
+            validated_retry_data = retry_defaults
```

Comment:
This should be something like
```python
elif isinstance(retry_data, bool):
    if retry_data:
        validated_retry_data = retry_defaults
    else:
        log.debug("State is set with explicit retry: False so using default retry configuration with 0 attempts")
        retry_defaults = {
            "until": True,
            "attempts": 0,
            "splay": 0,
            "interval": 30,
        }
```
We may (or may not) want to handle `retry_data is None`

Reply:
works for me, thanks